app/Filters/ApiPermissionFilter.php

```
@@ -0,0 +1,50 @@
+<?php
+
+namespace App\Filters;
+
+use CodeIgniter\Filters\FilterInterface;
+use CodeIgniter\HTTP\RequestInterface;
+use CodeIgniter\HTTP\ResponseInterface;
+
+class ApiPermissionFilter implements FilterInterface
+{
+    public function before(RequestInterface $request, $arguments = null): ResponseInterface|null
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$request'. (undefined)

(UnusedFormalParameter)